src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -405,6 +405,39 @@ static void M()
             Assert.Equal(attrType, info.Symbol);
         }
 
+        [Fact]
+        public void LocalFunctionAttribute_SpeculativeSemanticModel()
+        {
+            var text = @"
+using System;
+
+class A : Attribute { }
+
+class C
+{
+    void M()
+    {
+        void local1() { }
+    }
+}
+";
+            var tree = SyntaxFactory.ParseSyntaxTree(text, TestOptions.RegularPreview);
+            var compilation = CreateCompilation(tree);
+            var model = compilation.GetSemanticModel(tree);
+
+            var newTree = SyntaxFactory.ParseSyntaxTree(text.Replace("void local1", "[A] void local1"), TestOptions.RegularPreview);
+            var localFunctionSyntax = newTree.GetRoot()
+                .DescendantNodes()
+                .OfType<LocalFunctionStatementSyntax>()
+                .Single();
+            var method = newTree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Single();
+            Assert.True(model.TryGetSpeculativeSemanticModelForMethodBody(method.Body.SpanStart, method, out var speculativeModel));
+
+            var symbol = speculativeModel.GetDeclaredSymbol(localFunctionSyntax).GetSymbol<LocalFunctionSymbol>();
```

Comment:
What happens if you ask the same thing using the regular semantic model?

Reply:
The old model throws an exception if it is passed the new syntax node.

Reply:
Right, I meant if you constructed the final tree, then asked the question using a new semantic model.

Reply:
Got it. I'll add it to the test.

Reply:
Done